pkg/reduce/data_forward_test.go

```diff
@@ -647,7 +647,7 @@ func TestReduceDataForward_IdleWM(t *testing.T) {
 	assert.Equal(t, isb.WMB, msgs[0].Kind)
 	// the second message should be the data message from the closed window above
 	// in the test ApplyUDF above we've set the final message to have key="result"
-	for len(msgs[1].Keys) == 0 {
+	for len(msgs) < 2 || len(msgs[1].Keys) == 0 {
```

Comment:
what made you add this `len(msgs) < 2`, check?

Reply:
There was a bug in simple in memory buffer implementation, because of which this test was working fine without this check.  https://github.com/numaproj/numaflow/blob/491b6374b8ce0312eca08110ccc723acea863347/pkg/isb/stores/simplebuffer/buffer.go#L269